tutorials/src/bin/basic-tutorial-3.rs

```diff
@@ -0,0 +1,108 @@
+extern crate gstreamer as gst;
+
+use gst::prelude::*;
+
+fn main() {
+    // Initialize GStreamer
+    gst::init().unwrap();
+
+    // Create the elements
+    let source = gst::ElementFactory::make("uridecodebin", "source")
+        .expect("Could not create uridecodebin element.");
+    let convert = gst::ElementFactory::make("audioconvert", "convert")
+        .expect("Could not create convert element.");
+    let sink = gst::ElementFactory::make("autoaudiosink", "sink")
+        .expect("Could not create sink element.");
+
+    // Create the empty pipeline
+    let pipeline = gst::Pipeline::new("test-pipeline");
+
+    // Build the pipeline Note that we are NOT linking the source at this
+    // point. We will do it later.
+    pipeline.add_many(&[&source, &convert, &sink]).unwrap();
+    convert.link(&sink).expect("Elements could not be linked.");
+
+    // Set the URI to play
+    let uri = "https://www.freedesktop.org/software/gstreamer-sdk/data/media/sintel_trailer-480p.webm";
+    source.set_property_from_str("uri", uri);
+
+    // Connect the pad-added signal
+    let pipeline_clone = pipeline.clone();
+    let convert_clone = convert.clone();
+    source.connect_pad_added(move |_, src_pad| {
+        let pipeline = &pipeline_clone;
+        let convert = &convert_clone;
+
+        println!("Received new pad {} from {}",
+                 src_pad.get_name(),
+                 pipeline.get_name());
+
+        let sink_pad = convert.get_static_pad("sink")
+            .expect("  Failed to get static sink pad from convert");
+        if sink_pad.is_linked() {
+            println!("  We are already linked. Ignoring.");
```

Comment:
Why the spaces at the front of the strings?

Reply:
I was just copying the strings from the C examples, they also have some spaces in front. Removed them now.

Reply:
Want to send a patch for the C ones too?